src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -48,7 +48,7 @@
 
 @Log4j2
 @ToolAnnotation(CreateAlertTool.TYPE)
-public class CreateAlertTool implements Tool {
+public class CreateAlertTool implements WithModelTool {
```

Comment:
nit: instead of have a interface `WithModelTool`, i would recommend have another interface `ModelAwareResource`

```java

interface ModelAwareResource {
   String modelId();
}

public class CreateAlertTool implements Tool, ModelAwareResource {
```